io/eolearn/tests/test_io.py

```diff
@@ -321,7 +345,192 @@ def test_feature_dimension(self):
+                feature_type=FeatureType.DATA,
+                feature_name='BANDS-S2-L1C',
+                request=S2L1CWCSInput(
+                    layer='BANDS-S2-L1C',
+                    resx=resx,
+                    resy=resy,
+                    instance_id=instance_id
+                )
+            ),
+            cls.ImageFormatTestCase(
+                name='L8L1CWCSInput',
+                feature_type=FeatureType.DATA,
+                feature_name='TRUE-COLOR-L8',
+                request=L8L1CWCSInput(
+                    layer='TRUE-COLOR-L8',
+                    resx=resx,
+                    resy=resy,
+                    instance_id=instance_id
+                )
+            ),
+            cls.ImageFormatTestCase(
+                name='Sen2CorClassificationFeature - CLD',
+                feature_type=FeatureType.DATA,
+                feature_name='CLD',
+                request=AddSen2CorClassificationFeature(
+                    sen2cor_classification='CLD',
+                    layer='BANDS-S2-L2A',
+                    service_type=ServiceType.WCS,
+                    size_x=resx,
+                    size_y=resy,
+                    instance_id=instance_id
+                )
+            ),
+            cls.ImageFormatTestCase(
+                name='AddSen2CorClassificationFeature - SNW',
+                feature_type=FeatureType.DATA,
+                feature_name='SNW',
+                request=AddSen2CorClassificationFeature(
+                    sen2cor_classification='SNW',
+                    layer='BANDS-S2-L2A',
+                    service_type=ServiceType.WCS,
+                    size_x=resx,
+                    size_y=resy,
+                    instance_id=instance_id
+                )
+            ),
+            cls.ImageFormatTestCase(
+                name='AddSen2CorClassificationFeature - SCL',
+                feature_type=FeatureType.MASK,
+                feature_name='SCL',
+                request=AddSen2CorClassificationFeature(
+                    sen2cor_classification='SCL',
+                    layer='BANDS-S2-L2A',
+                    service_type=ServiceType.WCS,
+                    size_x=resx,
+                    size_y=resy,
+                    instance_id=instance_id
+                )
+            ),
+            # cls.ImageFormatTestCase(
+            #     name='DEMWCSInput',
+            #     feature_type=FeatureType.MASK_TIMELESS,
+            #     feature_name='DEM',
+            #     request=DEMWCSInput(
+            #         layer='DEM',
+            #         resx=resx,
+            #         resy=resy,
+            #         instance_id=instance_id
+            #     )
+            # )
+
+        ]
+
+        cls.dem_test = [
+        ]
+
+    def test_image_format(self):
+        for test in self.tests:
+            with self.subTest(msg='Test case {}'.format(test.name)):
+                if test.feature_type == FeatureType.DATA:

```

Comment:
2 comments here:

- You have 2 blocks of code in (if statements) that are almost duplicated. You could do:
```Python
atol_value = 1e-4 if test.feature_type is FeatureType.DATA else 0
```
or even nicer:
```Python
atol_value = 0 if test.feature_type.is_discrete() else 1e-4
```
and then run `np.testing.assert_allclose`

- For the future: it is more correct to compare 2 objects (e.g. Enums) with `is` instead of `==`. The `==` is meant for comparing values.

Reply:
👍